recipes/open62541/all/conanfile.py

```diff
@@ -0,0 +1,238 @@
+    topics = (
+        "OPC UA", "open62541", "sdk", "server/client", "c", "iec-62541",
+        "industrial automation", "tsn", "time sensetive networks", "publish-subscirbe", "pubsub"
+    )
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "fPIC": [True, False],
+        "shared": [True, False],
+        "historize": [True, False],
+        "logging_level": ["Fatal", "Error", "Warning", "Info", "Debug", "Trace"],
+        "subscription": [True, False],
+        "subscription_events": [True, False],
+        "methods": [True, False],
+        "dynamic_nodes": [True, False],
+        "single_header": [True, False],
+        "multithreading": [True, False],
+        "imutable_nodes": [True, False],
+        "web_socket": [True, False],
+        "discovery": [True, False],
+        "discovery_semaphore": [True, False],
+        "discovery_multicast": [True, False],
+        "query": [True, False],
+        "encription": [True, False],
+        "json_support": [True, False],
+        "pub_sub": ["None", "Simple", "Ethernet", "Ethernet_XDP"],
+        "data_access": [True, False],
+        "compiled_nodeset_descriptions": [True, False],
+        "namescpae_zero": ["MINIMAL", "REDUCED", "FULL"],
+        "embedded_profile": [True, False],
+        "typenames": [True, False],
+        "hardening": [True, False],
+        "cpp_compatible": [True, False],
+        "readable_statuscodes": [True, False]
+    }
+    default_options = {
+        "fPIC": True,
+        "shared": True,
+        "historize": False,
+        "logging_level": "Info",
+        "subscription": True,
+        "subscription_events": False,
+        "methods": True,
+        "dynamic_nodes": True,
+        "single_header": False,
+        "multithreading": False,
+        "imutable_nodes": False,
+        "web_socket": False,
+        "discovery": True,
+        "discovery_semaphore": True,
+        "discovery_multicast": False,
+        "query": False,
+        "encription": False,
+        "json_support": False,
+        "pub_sub": "None",
+        "data_access": True,
+        "compiled_nodeset_descriptions": True,
+        "namescpae_zero": "FULL",
+        "embedded_profile": False,
+        "typenames": True,
+        "hardening": False,
+        "cpp_compatible": False,
+        "readable_statuscodes": True
+    }
+    generators = "cmake", "cmake_find_package"
+
+    def requirements(self):
+        if self.options.encription:
+            self.requires("mbedtls/2.16.3-gpl")
```

Comment:
Are other encryption libraries possible? Like openssl?
If so, I'd change the option encryption to receive a list of possible encryption packages

Reply:
openssl is possible with version 1.1 (see my pr).
With mbedtls, why forcing gpl variant ?

Reply:
This specific commit is focusing on 1.0 and 1.0.1 versions which do not support OpenSSL as an option, I started this PR before 1.1 was even released as a Release Candidate and want to finish that before adding support for v1.1 (which won't be that hard once this is done, and yes I do plan to add OpenSSL as an option) 

Regarding [mbedtls](https://github.com/conan-io/conan-center-index/blob/master/recipes/mbedtls/all/conanfile.py) and GPL-2.0 or Apache-2.0 - I am not aware of any difference in  code apart of the different Licenses and frankly I do not understand the need for two different licenses, I simply chose the one I had installed on my machine. I could provide an option for the user to choose one over the other, but does it change anything significant in the code base, which would warrant the added option? If not I would stay on the side of simplicity and skip it.

Reply:
I'd prefer to go for the less restrictive license here

Reply:
I'll change it to Apache then